src/app/services/gateway.service.ts

```diff
@@ -16,10 +18,17 @@ export class GatewayService {
         return this._httpClient.get<PostResponse>(this._data);
     }
 
-    postRegistrationForm(user: User): Observable<User> {
+    postRegistrationForm(user: User): Observable<Token | Error> {
         return this._httpClient.post(
             'http://localhost:3000/api/v1/auth/sign-in',
             user,
-        ) as Observable<User>;
+        ) as Observable<Token>;
```

Comment:
The typing at the end does not correspond to typing of the function. Why did you omit `Error`?